### ISSUE TITLE
UMLS concept ID mappig

### ISSUE THREAD

**Issue by przemokraw** [chanzuckerberg/MedMentions#5]
First of all I would like to thank you for sharing this dataset. It's great!

I am facing the following problem:
I have a NER and entity linker that links genes to NCBI GeneID (eg. TP53  has GeneID 7157, see https://www.ncbi.nlm.nih.gov/gene/?term=tp53), drugs to ChEMBL id (eg. https://www.ebi.ac.uk/chembl/compound_report_card/CHEMBL189707/) and diseases to Disease Ontology ID (http://disease-ontology.org/). I need to link my entities to corresponding UMLS concept IDs in order to evaluate my NER. Disease Ontology has an API that returns CUI for given DOID. Unfortunately, I don't know how to deal with genes and drugs. Please, give a hint if you know how to map NCBI GeneID or ChEMBL id to UMLS concept ID in reliable way.

Thank you for your great work and any help!

**Comment by czi-sunil** [comment]
I have not mapped from these ontologies, but was able to use the information in the MRCONSO.RRF file (part of the UMLS release) to map between MeSH and UMLS ids.

**Comment by sandertan** [comment]
Old thread, but perhaps this could help:

UMLS contains the HGNC gene dataset, so there is a mapping between CUI and GeneID. If you browse MRCONSO, you'll find the gene names and NCBI IDs and their respective UMLS CUI.

See:
https://www.nlm.nih.gov/research/umls/sourcereleasedocs/current/HGNC/stats.html
https://www.genenames.org/download/statistics-and-files/

**Comment by island99** [comment]
In the paper, it says: 
```
entity descriptions (which in our case is the canonical name, the type, and a list of known aliases andsynonyms)
```
Is there any examples of entity descriptions to show? I'm not sure how to get the above and what they mean, thanks for the help.

**Comment by kaushikacharya** [comment]
> Is there any examples of entity descriptions to show? I'm not sure how to get the above and what they mean, thanks for the help.

You can have a look at Metathesaurus Browser mentioned in https://www.nlm.nih.gov/research/umls/index.html
N.B. You would need to register first.

Here's an example screenshot:
![image](https://user-images.githubusercontent.com/4562867/103474032-71ea2b00-4dc5-11eb-82f4-ff63cb1ba84c.png)

**Explanation**:
**Canonical Name**: Nodule of lung

**Aliases**:
- pulmonary nodule
- lung nodule
The above two are a subset of the aliases (mentioned under **Atoms**) for this entity.

**Comment by srishti-git1110** [comment]
Hi,
A very old thread, but I'm struggling to find a solution to a problem of mine and this thread looks very relevant. Please help me.

My question- I need to map some disease names (that I've tagged using a pre-trained tool) to MeSH+OMIM ontology as MeSH doesn't have genetic diseases, so any complete ontology for diseases shall use both MeSh + OMIM.

I have the CUIs corresponding to those diseases & would now like to map the CUI ids to MeSH+OMIM. 

What are the possible solutions? Thanks in advance.

**Comment by czi-sunil** [comment]
UMLS contains mappings to source vocabularies.
If you go to the UMLS Metathesaurus Browser and look up C5203670, which is
the id for "COVID-19 (disease)", you will see a list of names from a bunch
of source Vocabularies that are mapped to this concept. "MSH" is MeSH and
"OMIM" is OMIM.
This information is also in the MRCONSO.RRF file.


*...............................Sunil Mohan*

Research Scientist


***@***.***
http://chanzuckerberg.com <http://chanzuckerberg.com/?email>

**Comment by srishti-git1110** [comment]
Thanks for answering.

I'll need to perform this mapping from cui to mesh+omim for quite a lot of data & need to integrate this into my code pipeline.
Do you have any idea how I could do this?

I tried to download the MRCONSO.RRF file (with a hope to accomplish my task using information from this file) but it requires some sort of license.

**Comment by czi-sunil** [comment]
Yes, you will need to get permission to download a UMLS release (I think
you need to create an account, and describe your use). MRCONSO.RRF is part
of the download. That is probably the best way to proceed.


*...............................Sunil Mohan*

Research Scientist


***@***.***
http://chanzuckerberg.com <http://chanzuckerberg.com/?email>

**Comment by Youpu-Chen** [comment]
MedGen Database may be a manageable solution.

**Comment by Youpu-Chen** [comment]
MedGen Database may be a managable solution.
